### ISSUE TITLE
Confirming Implementation of URLBASE and BASEPATH

### ISSUE THREAD

**Issue by brettpalmberg** [CrunchyData/pg_featureserv#119]
## Question / Bug?

Hosting pg_featureserv at a non-root path required including the path parameter (e.g. `/features`) in both the `UrlBase` and `BasePath` config params to get URLs and JSON values to form correctly. It felt redundant to have to include the BasePath in both variables, but wanted to confirm whether this is the intended behavior before calling anything a bug.

### Example (this works)

To host pg_featureserv at `localhost:9000/features/`, the following environment variables are required to make all .html links and JSON-built urls work correctly:

- PGFS_SERVER_URLBASE=http://localhost:9000/features
- PGFS_SERVER_BASEPATH=/features

### What I expected (this does not work)

- PGFS_SERVER_URLBASE=http://localhost:9000
- PGFS_SERVER_BASEPATH=/features

Thank you.

**Comment by brettpalmberg** [comment]
Related to #109. I was able to confirm that the behavior of URLBASE and BASEPATH configuration parameters behave differently between `pg_tileserv` and `pg_featureserv`. `pg_tileserv` behaves as described in the section _What I expected (this does not work)_ above.

**Comment by dr-jts** [comment]
Thanks, @brettpalmberg .  I agree that having no redundancy between `URLBASE` and `BASEPATH` is desirable.  And so is compatibility between `pg_featureserv` and `pg_tileserv`.  We'll investigate making this change.

**Comment by brettpalmberg** [comment]
Excellent, thanks for your response @dr-jts! Feel free to @ me if there's anything I can contribute. I wasn't familiar enough with the codebase just yet to be comfortable jumping-in :)

**Comment by rhewy** [comment]
Can these environment variables be passed to the docker container?

**Comment by pramsey** [comment]
Yes! Note that a `Configuration.Variable` will get converted into a `PGFS_CONFIGURATION_VARIABLE` if you catch my drift. There's a standard prefix (`PGFS_`) and everything gets caps'ed and underscored.

**Comment by rhewy** [comment]
So in a docker env file would I?
PGFS_SERVER_URLBASE=https://vm103.athena.bcit.ca/pgfs
PGFS_SERVER_BASEPATH=/pgfs

or

SERVER_URLBASE=https://vm103.athena.bcit.ca/pgfs
SERVER_BASEPATH=/pgfs

**Comment by pramsey** [comment]
The former

**Comment by rhewy** [comment]
So, I have nginx as the reverse proxy for https
        location /pgfs/ {
             proxy_pass http://localhost:9000/ ;
        }

Then https://vm103.athena.bcit.ca/pgfs is a 404 
However
https://vm103.athena.bcit.ca/pgfs/pgfs/
Works

Then from the links on the collections page:
https://vm103.athena.bcit.ca/pgfs/pgfs/collections.html

The links are missing the second /pgfs/
https://vm103.athena.bcit.ca/pgfs/collections/public.becpas.html
404 error for the above

I will try some things ...

**Comment by rhewy** [comment]
This worked for me:

PGFS_SERVER_URLBASE=https://vm103.athena.bcit.ca/pgfs
PGFS_SERVER_BASEPATH=/

Home links work too

Thank you!
